src/components/CreateOrModifyTagDialog.tsx

```diff
@@ -48,13 +48,14 @@ export default function CreateOrModifyTagDialog({
 		useCreateOrModifyTag(currentTagId, form.reset);
 
 	useEffect(() => {
-		if (tagDetails) {
-			form.setValue('name', tagDetails.name);
-			tagDetails.color && form.setValue('color', tagDetails.color);
-			tagDetails.description &&
-				form.setValue('description', tagDetails.description);
+		if (isModalOpen && modalState === 'EDIT_TAG' && tagDetails) {
```

Comment:
I guess you can creat one function that will be reset form. E.g.: handleEditTagFormReset() and handleAddTagFormReset.

To discuss, because I do not understand the business context

Reply:
This is the context: User can create new tag and also modify tag. When he adds new tag, user can fill all fields, but when he accidently close modal, he can reopen it again with no data lost. When he edit tag - the form should be displayed as new, with filled values from the api.